### ISSUE TITLE
False positive for `valid-title` rule with test.each

### ISSUE THREAD

**Issue by ericcornelissen** [jest-community/eslint-plugin-jest#665]
## The Problem

With a recent updated I noticed that the rule [`jest/valid-title`](https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/valid-title.md) triggers for something like the following snippet (I hadn't noticed it before because the rule used to be disabled by default).

```js
const testNumbers = test.each([1, 2, 3]);

testNumbers("works for %d", (numbers) => {
  ... // Test here, clearly we _could_ just use `test.each([1, 2, 3])("works for %d", () => {});` here...
});
```

I don't know if this is standard practice, but I sometimes use it if I need the same `test.each` for a couple of tests (I know I can wrap them in a `describe.each` but sometimes that doesn't feel right to me) or when the contents of `test.each` is rather big.

Anyway, this snippets does have a "valid-title" but it is reported as being invalid on line 1. I'm guessing due to the fact that on line 1 I'm simply not providing a title. However, I'm also not starting a test on line 1.

I have not exhaustively tried all rules to see if this happens with any other rules.

## Proposed Solution

I'm not sure how easy it is to detect this but the title of a `test.each` should only be checked if it is actually followed by a test case. Ideally, in the snippet above, it would even detect the subsequent call to `testNumbers` and check the validity of the title there, but that might be too much to ask.

**Comment by G-Rath** [comment]
I'm somewhat against supporting this, as you can just do:

```
const testNumbers = [1, 2, 3];

test.each(testNumbers)("works for %d", (numbers) => {
  ... // Test here, clearly we _could_ just use `test.each([1, 2, 3])("works for %d", () => {});` here...
});
```

There becomes a point where supporting all of the ways results in either us building a static type system (for which there is TypeScript), or weakening rules because of edge cases (i.e because we can't support all cases without being a static type system, so we allow users to shoot themselves in the foot easier).

**Comment by ericcornelissen** [comment]
Thanks for the quick response @G-Rath, I was somewhat hesitant submitting this issue because I expected an answer along these lines (but submitted it anyway in case of a different answer).

I totally understand your point and am happy to use your suggestion if that means being able to benefit from the "valid-title" rule :+1:
